ytmusicapi/mixins/browsing.py

```diff
@@ -239,23 +240,23 @@ def get_artist(self, channelId: str) -> Dict:
         artist.update(self.parser.parse_artist_contents(results))
         return artist
 
-    def get_artist_albums(self, channelId: str, params: str) -> List[Dict]:
+    def get_artist_albums(self, channelId: str, params: str) -> Optional[List[Dict]]:
         """
         Get the full list of an artist's albums or singles
 
         :param channelId: channel Id of the artist
         :param params: params obtained by :py:func:`get_artist`
-        :return: List of albums in the format of :py:func:`get_library_albums`,
+        :return: Optionally a List of albums in the format of :py:func:`get_library_albums`,
           except artists key is missing.
 
         """
         body = {"browseId": channelId, "params": params}
         endpoint = 'browse'
         response = self._send_request(endpoint, body)
-        results = nav(response, SINGLE_COLUMN_TAB + SECTION_LIST_ITEM + GRID_ITEMS)
-        albums = parse_albums(results)
-
-        return albums
+        results = nav(response, SINGLE_COLUMN_TAB + SECTION_LIST_ITEM + GRID_ITEMS, True)
+        if results:
+            albums = parse_albums(results)
+            return albums
```

Comment:
Surely having a guaranteed returned type is safer? If you expect it to always return a List, even if it's not useful (empty), at least it's consistent

Reply:
I've changed it